index.js

```diff
@@ -2,9 +2,14 @@
 //
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
 
+const path = require('path')
+const parse = require('module-details-from-path')
+const { fileURLToPath } = require('url')
+
 const importHooks = [] // TODO should this be a Set?
 
 const setters = new WeakMap()
+const specifiers = new Map()
```

Comment:
Not without future-proofing for loaders-in-threads. The only reason `_register` is even called from the generated code below is because loaders-in-threads may happen.

Reply:
Threads can't share object (or functions).